From 5ee6f49297c8137cae527429e0267462c14ec3ed Mon Sep 17 00:00:00 2001
From: cristy <cristy@aa41f4f7-0bf4-0310-aa73-e5a19afd5a74>
Date: Tue, 23 Dec 2014 01:25:22 +0000
Subject: [PATCH] During identification of image do not fill memory

This create a security risk (DOS) by filling all memory during identification of image.

git-svn-id: https://subversion.imagemagick.org/subversion/ImageMagick/branches/ImageMagick-6@17413 aa41f4f7-0bf4-0310-aa73-e5a19afd5a74
origin: http://trac.imagemagick.org/changeset/17413
---
 coders/tiff.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/coders/tiff.c b/coders/tiff.c
index 1fe12fe..79033a6 100644
--- a/coders/tiff.c
+++ b/coders/tiff.c
@@ -1134,7 +1134,8 @@ RestoreMSCWarning
       SetImageColorspace(image,CMYKColorspace);
     if (photometric == PHOTOMETRIC_CIELAB)
       SetImageColorspace(image,LabColorspace);
-    TIFFGetProfiles(tiff,image);
+    if (image_info->ping == MagickFalse)
+      TIFFGetProfiles(tiff,image);
     TIFFGetProperties(tiff,image);
     option=GetImageOption(image_info,"tiff:exif-properties");
     if ((option == (const char *) NULL) ||
-- 
2.1.4

